rpcs3/Emu/Cell/Modules/cellSaveData.cpp

```diff
@@ -141,8 +141,15 @@ static NEVER_INLINE s32 savedata_op(ppu_thread& ppu, u32 operation, u32 version,
 						save_entry2.atime = entry.atime;
 						save_entry2.mtime = entry.mtime;
 						save_entry2.ctime = entry.ctime;
-						//save_entry2.iconBuf = NULL; // TODO: Here should be the PNG buffer
-						//save_entry2.iconBufSize = 0; // TODO: Size of the PNG file
+						if (fs::is_file(base_dir + entry.name + "/ICON0.PNG"))
+						{
+							fs::file icon = fs::file(base_dir + entry.name + "/ICON0.PNG");
+							u32 iconSize = icon.size();
+							uchar* iconData = new uchar[iconSize];
+							icon.read(iconData, iconSize);
```

Comment:
What would you do with the struct?  I concede the std::shared_ptr<void> isn't very clean. But, using a vector would mean the vector gets copied when save_entries.push_back is called.  To me, it's not ideal copying data and immediately deleting the other copy. I'm welcome to suggestions as I would probably do a std::shared_ptr<uchar> myself or maybe a std::shared_ptr<std::vector<uchar>>, but the original comment had a void* pointer, and I tried to respect that.